src/utils.js

```diff
@@ -0,0 +1,38 @@
+export const extractParams = route => {
```

Comment:
@lukejacksonn Query is nice, but I recommend we work on this after the new  router is published, so that it's done right and better researched.

Reply:
What is the advantage of doing this later and what is your concern with this implementation?

I'd argue any querystring parsing is better than none. I can clearly document which cases this will work for and for which it will fail. If anyone wants to propose an improvement it then they can once this gets merged.

Reply:
@lukejacksonn You are moving too fast. When you move fast things get out of control quickly. Users will come in asking for new features before we are even sure we even need this or not. Then we might break the API. If we push only the bare minimum we'll be safe.

Reply:
@lukejacksonn I am not sure we need a qs in the router.

Reply:
In my opinion the router needs to handle query string parsing.  At the very least, allow users to pass in their own function as a configuration option.

Reply:
@JorgeBucaran Was there a specific reason you didn't feel a qs parser was needed in the router?

Reply:
Not particularly, I was just trying to simplify the PR so we could merge it more quickly. I'll have a look at this file and leave comments then. 👍